Public/Src/Tools/SymbolDaemon/Tool.SymbolDaemonInterfaces.dsc

```diff
@@ -0,0 +1,162 @@
+    warningTimeoutInMilliseconds?: number;
+
+    /** ------------------------------------------------------------------- */
+    
+    /** Number of retries to connect to a running SymbolDaemon process. */
+    maxConnectRetries?: number;
+    
+    /** Delay between retries to connect to a running SymbolDaemon process. */
+    connectRetryDelayMillis?: number;
+    
+    /** Request name. */
+    name?: string;
+    
+    /** Symbol service config file. */
+    symbolServiceConfigFile?: File;
+    
+    /** Where to save console output. */
+    consoleOutput?: Path;
+}
+
+/**
+ * VSO Symbol settings.
+ */
+@@public
+export interface SymbolRequestSettings {
+    /** Service URL. */
+    service?: string;
+
+    /** Request name. */
+    name?: string;
+    
+    /** Maximum number of uploads to issue to symbol service in parallel. */
+    maxParallelUploads?: number;
+    
+    /** Retention period in days for uploaded symbols. */
+    retentionDays?: number;
+    
+    /** Enable symbol telemetry. */
+    enableTelemetry?: boolean;
+}
+
+/**
+ * SymbolDaemon settings
+ */
+@@public
+export interface DaemonSettings {
+    /** Maximum number of clients SymbolDaemon should process concurrently. */
+    maxConcurrentClients?: number;
+    
+    /** Whether SymbolDaemon should send ETW events about the progress of symbol
+     *  operations for the purpose of integration with CloudBuild. */
+    enableCloudBuildIntegration?: boolean;
+    
+    /** Verbose logging. */
+    verbose?: boolean;
+}
+
+/**
+ * Arguments for starting the SymbolDaemon service.
+ */
+@@public
+export interface ServiceStartArguments extends DaemonSettings, OperationArguments {}
+
+/**
+ * Arguments for the 'symbold create' operation.
+ */
+@@public
+export interface SymbolCreateArguments extends SymbolRequestSettings, DaemonSettings, OperationArguments {}
+
+/** 
+ * Common properties for all 'result' types.
+ */
+@@public
+export interface Result {
+    /** All required outputs (so that other pips can take those as dependencies). */
+    outputs: DerivedFile[];
+}
+
+/**
+ * Information about started symbold service (returned as part of ServiceStartResult)
+ */
+@@public
+export interface ServiceStartResult extends Result {
+    symbolDaemonId: Transformer.ServiceId;
+    ipcMoniker: IpcMoniker;
+}
+
+/**
+ * Result for the 'symbold create' operation.
+ */
+@@public
+export interface SymbolCreateResult extends Result {
+    /** Info about the started service */
+    serviceStartInfo: ServiceStartResult;
+}
+
+/**
+ * Operations provided by a runner.
+ */
+@@public
+export interface SymbolRunner {
+    /** Invokes 'symbolc create'. */
+    createSymbol: (args: SymbolCreateArguments) => SymbolCreateResult;
+
+    /** 
+     * Adds files to symbol. 
+     * Preferred method is to use addArtifactsToSymbol.
+     */
+    addFilesToSymbol: (createResult: SymbolCreateResult, args: OperationArguments, fileInfos: File[]) => Result;

```

Comment:
again suprized not to see the ability to link to source control in the arguments

Reply:
see my reply to a similar comment

---
In reply to: [313492200](https://github.com/microsoft/BuildXL/pull/727#discussion_r313492200) [](ancestors = 313492200)